wilsonChan/assignments/slidingwindow/lc209/lc209.py

```diff
@@ -1,8 +1,30 @@
+# Problem Statement #
+
+# Given an array of positive numbers and a positive number ‘S,’ find the length of the smallest contiguous subarray whose sum is greater than or equal to ‘S’. Return 0 if no such subarray exists.
+
 import math
 
 
 def smallest_subarray_with_given_sum(s, arr):
 
+  #Initialized variables
+  window_start = 0        #left/beginning of the window
+  window_end = 0          #right/end of the window
+  current_sum = 0
+  min_length = len(arr) + 1
+
+  #Window continues moving to the right as long it's less than whole array
+  while window_end < len(arr):
+    current_sum += arr[window_end]
+    
+    while current_sum >= s:
+      min_length = min(min_length, window_end - (window_start + 1))
+      current_sum -= arr[window_start]
+      window_start += 1
+
+    window_end += 1
+
+  return current_sum if current_sum <= len(arr) else 0
```

Comment:
I think you are going to want to return the minimum length, not the current sum.